pkg/controllers/provisioningv2/fleetcluster/fleetcluster_test.go

```diff
@@ -0,0 +1,236 @@
+		{
+			Key:      "key",
+			Operator: corev1.TolerationOpEqual,
+			Value:    "value",
+		},
+	}
+)
+
+func TestCluster(t *testing.T) {
+	require := require.New(t)
+
+	h := &handler{
+		clustersCache:     &fakeClusterCache{},
+		getPrivateRepoURL: func(*v1.Cluster, *mgmt.Cluster) string { return "" },
+	}
+
+	cluster := &v1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster", Namespace: "test-namespace",
+		},
+		Spec: v1.ClusterSpec{},
+	}
+	clusterStatus := v1.ClusterStatus{ClusterName: "cluster-name", ClientSecretName: "client-secret-name"}
+
+	labels := map[string]string{"cluster-group": "cluster-group-name"}
+
+	tests := []struct {
+		name          string
+		cluster       *v1.Cluster
+		status        v1.ClusterStatus
+		clustersCache mgmtv3.ClusterCache
+		wantLen       int
+		wantFleet     *fleet.Cluster
+	}{
+		{
+			"cluster-has-no-customization",
+			cluster,
+			clusterStatus,
+			newClusterCache(map[string]*mgmt.Cluster{
+				"cluster-name": newMgmtCluster(
+					"cluster-name",
+					labels,
+					mgmt.ClusterSpec{},
+				),
+			}),
+			1,
+			&fleet.Cluster{
+				Spec: fleet.ClusterSpec{
+					AgentAffinity: &builtinAffinity,
+				},
+			},
+		},
+		{
+			"cluster-has-affinity-override",
+			cluster,
+			clusterStatus,
+			newClusterCache(map[string]*mgmt.Cluster{
+				"cluster-name": newMgmtCluster(
+					"cluster-name",
+					labels,
+					mgmt.ClusterSpec{
+						ClusterSpecBase: mgmt.ClusterSpecBase{
+							FleetAgentDeploymentCustomization: &mgmt.AgentDeploymentCustomization{
+								OverrideAffinity:             &linuxAffinity,
+								OverrideResourceRequirements: &corev1.ResourceRequirements{},
+								AppendTolerations:            []corev1.Toleration{},
+							},
+						},
+					},
+				),
+			}),
+			1,
+			&fleet.Cluster{
+				Spec: fleet.ClusterSpec{
+					AgentAffinity:    &linuxAffinity,
+					AgentResources:   &corev1.ResourceRequirements{},
+					AgentTolerations: []corev1.Toleration{},
+				},
+			},
+		},
+		{
+			"cluster-has-custom-tolerations-and-resources",
+			cluster,
+			clusterStatus,
+			newClusterCache(map[string]*mgmt.Cluster{
+				"cluster-name": newMgmtCluster(
+					"cluster-name",
+					labels,
+					mgmt.ClusterSpec{
+						ClusterSpecBase: mgmt.ClusterSpecBase{
+							FleetAgentDeploymentCustomization: &mgmt.AgentDeploymentCustomization{
+								OverrideAffinity:             nil,
+								OverrideResourceRequirements: resourceReq,
+								AppendTolerations:            tolerations,
+							},
+						},
+					},
+				),
+			}),
+			1,
+			&fleet.Cluster{
+				Spec: fleet.ClusterSpec{
+					AgentAffinity:    &builtinAffinity,
+					AgentResources:   resourceReq,
+					AgentTolerations: tolerations,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h.clustersCache = tt.clustersCache
+			objs, _, err := h.createCluster(tt.cluster, tt.status)
+			require.Nil(err)
+			require.NotNil(objs)
+
+			require.Equal(tt.wantLen, len(objs))
```

Comment:
I think this is unnecessary and would just do

`require.Equal(1, len(objs))` and not pass `tt.wantLen` through the test params because it's the same for all tests and will reduce your code.